### PR TITLE
FlutterFire plugins & SDK updates (branch: begin)

### PR DESCRIPTION
**Pull request by mzdm**
FlutterFire plugins were recently updated with breaking changes so I think the project should be updated to the newest versions which provide better stability and maintainability.

- Changed Android target API from 28 to 29
- Also contains a fix to #3

### CODE REVIEW

**Comment by ditman** [comment]
@mzdm I had been meaning to do this myself, thanks for sending the PR! I'll review and merge by EOD today!

File: lib/main.dart [hunk 1]
```diff
@@ -12,12 +12,17 @@
...
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 
 import 'src/app.dart' deferred as app;
 
-void main() {
-  final Future<void> loadedLibrary = app.loadLibrary();
+void main() async {
+  // Initialize Firebase
+  WidgetsFlutterBinding.ensureInitialized();
```
Comment (ditman) [inline lib/main.dart:22]: Is this required? I've only seen this in tests?

**Review by ditman — COMMENTED** [review]
Just one question about the `WidgetsFlutterBinding`

File: lib/main.dart (hunk 1 above)
Comment (mzdm) [reply lib/main.dart:22]: That implementation comes from [this SO answer](https://stackoverflow.com/a/63492262) (before I used [this from the offic docs](https://firebase.flutter.dev/docs/overview#initializing-flutterfire)). I think it ensures that the Flutter engine is activated so platform-specific code can be called

It is also used as an example [in pub.dev](https://pub.dev/packages/cloud_firestore/example)

Comment (ditman) [reply lib/main.dart:22]: Ah, ok, I understand now. I wasn't seeing that you're calling that in `main`, and not somewhere inside the App itself!

**Pull request merged**